Git-commit: 2f089cb89d2f47702c31bd584c12badc88bbe17c
From: Felipe Pena <felipensp@gmail.com>
Subject: block: xen-blkfront: Fix possible NULL ptr dereference
Patch-mainline: 3.13-rc3
References: bsc#961658 fate#320200

In the blkif_release function the bdget_disk() call might returns
a NULL ptr which might be dereferenced on bdev->bd_openers checking

Signed-off-by: Felipe Pena <felipensp@gmail.com>
Signed-off-by: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
[v2: Added WARN per Roger's suggestion]
Acked-by: jbeulich@suse.com

--- a/drivers/block/xen-blkfront.c
+++ b/drivers/block/xen-blkfront.c
@@ -2012,6 +2012,10 @@ static int blkif_release(struct gendisk
 
 	bdev = bdget_disk(disk, 0);
 
+	if (!bdev) {
+		WARN(1, "Block device %s yanked out from us!\n", disk->disk_name);
+		goto out_mutex;
+	}
 	if (bdev->bd_openers)
 		goto out;
 
@@ -2042,6 +2046,7 @@ static int blkif_release(struct gendisk
 
 out:
 	bdput(bdev);
+out_mutex:
 	mutex_unlock(&blkfront_mutex);
 	return 0;
 }
